File: src/main/java/com/scalag/vulkan/executor/BufferAction.java
package com.scalag.vulkan.executor;

import lombok.Data;

import static org.lwjgl.vulkan.VK10.VK_BUFFER_USAGE_TRANSFER_DST_BIT;
import static org.lwjgl.vulkan.VK10.VK_BUFFER_USAGE_TRANSFER_SRC_BIT;

@Data
public class BufferAction {
    public static final int
        DO_NOTHING = 0,
        LOAD_INTO = VK_BUFFER_USAGE_TRANSFER_DST_BIT,
        LOAD_FROM = VK_BUFFER_USAGE_TRANSFER_SRC_BIT;

    private final int action;
}

File: src/main/scala/com/scalag/vulkan/executor/BufferAction.scala
package com.scalag.vulkan.executor;

import org.lwjgl.vulkan.VK10.VK_BUFFER_USAGE_TRANSFER_DST_BIT;
import org.lwjgl.vulkan.VK10.VK_BUFFER_USAGE_TRANSFER_SRC_BIT;

enum BufferAction(val action: Int):
  case DO_NOTHING extends BufferAction(0)
  case LOAD_INTO extends BufferAction(VK_BUFFER_USAGE_TRANSFER_DST_BIT)
  case LOAD_FROM extends BufferAction(VK_BUFFER_USAGE_TRANSFER_SRC_BIT)
